ts/screens/services/ServiceDetailsScreen.tsx

```diff
@@ -215,42 +217,57 @@ class ServiceDetailsScreen extends React.Component<Props, State> {
       return (
         <React.Fragment>
           {metadata.description && (
-            <Markdown animated={true}>{metadata.description}</Markdown>
+            <Markdown
+              animated={true}
+              onLoadEnd={() =>
+                this.setState({
+                  isMarkdownLoaded: true
+                })
+              }
+            >
+              {metadata.description}
+            </Markdown>
           )}
           {metadata.description && <View spacer={true} large={true} />}
-          {metadata.tos_url &&
-            renderInformationLinkRow(
-              I18n.t("services.tosLink"),
-              metadata.tos_url
-            )}
-          {metadata.privacy_url &&
-            renderInformationLinkRow(
-              I18n.t("services.privacyLink"),
-              metadata.privacy_url
-            )}
-          {(metadata.app_android || metadata.app_ios || metadata.web_url) && (
-            <H4 style={styles.infoHeader}>
-              {I18n.t("services.otherAppsInfo")}
-            </H4>
+          {this.state.isMarkdownLoaded && (
```

Comment:
I'm pretty sure this code is little buggy: if **description** is **undefined** (description is an optional field in service metadata) **this.state.isMarkdownLoaded** is always false, then no other information will be rendered.
These items should be rendered also when description is undefined.

In addition you should consider to render conditionally to markodown load end also these included in **renderContactItems**

I test this login on this commit 617468e799c6d6f8d9db907601a2671ae69dee0f (here an additional comment 33725674da3f352581234798b65de8836dbb0af8)
What do you think? If it sounds good could you make some tests?

Reply:
ok starting from your commit , I integrated the changes with some improvements and added the test cases